src/vulkan-renderer/io/byte_stream.cpp

```diff
@@ -0,0 +1,112 @@
+#include "inexor/vulkan-renderer/io/byte_stream.hpp"
+#include "inexor/vulkan-renderer/world/cube.hpp"
+
+#include <fstream>
+
+namespace inexor::vulkan_renderer::io {
+std::vector<std::uint8_t> ByteStream::read_file(const std::filesystem::path &path) {
+    std::ifstream stream(path, std::ios::in | std::ios::binary);
+    return {std::istreambuf_iterator<char>(stream), std::istreambuf_iterator<char>()};
+}
+
+ByteStream::ByteStream(std::vector<std::uint8_t> buffer) : m_buffer(std::move(buffer)) {}
+
+ByteStream::ByteStream(const std::filesystem::path &path) : ByteStream(read_file(path)) {}
+
+std::size_t ByteStream::size() const {
+    return m_buffer.size();
+}
+[[nodiscard]] const std::vector<std::uint8_t> &ByteStream::buffer() const {
+    return m_buffer;
+}
+
+void ByteStreamReader::check_end(const std::size_t size) const {
+    if (std::distance(m_iter, m_stream.buffer().end()) < size) {
+        throw std::runtime_error("end would be overrun");
+    }
+}
+
+ByteStreamReader::ByteStreamReader(const ByteStream &stream) : m_stream(stream), m_iter(stream.buffer().begin()) {}
+
+void ByteStreamReader::skip(const std::size_t size) {
+    std::size_t skip = std::min(
+        size, std::size_t(std::distance<std::vector<std::uint8_t>::const_iterator>(m_iter, m_stream.buffer().end())));
+    std::advance(m_iter, skip);
+}
+
+std::size_t ByteStreamReader::remaining() const {
+    return std::distance<std::vector<std::uint8_t>::const_iterator>(m_iter, m_stream.buffer().end());
+}
+
+template <>
+std::uint8_t ByteStreamReader::read() {
+    check_end(1);
+    return *m_iter++;
+}
+
+template <>
+std::uint32_t ByteStreamReader::read() {
+    check_end(4);
+    return *m_iter++ | (*(m_iter++) << 8) | (*(m_iter++) << 16) | (*(m_iter++) << 24);
```

Comment:
```suggestion
    return (*m_iter++ << 0U) | (*m_iter++ << 8U) | (*m_iter++ << 16U) | (*m_iter++ << 24U);
```
Ok, so a few changes here. First of all, adding `U` suffix to all constants to avoid `use of signed integer with bitwise operator` clang-tidy warning. Second, removing the parens around the deref and increment (e.g. `*(m_iter++)` -> `*m_iter++`) since they're unneeded. Thirdly, adding `<< 0U` to the first expr, just to make it even and to silence the clang-tidy warning.